### DISCUSSION TITLE
Is there a way to set Dashboard Query Defaults?

### DISCUSSION THREAD

**Discussion by HarleeDavidson** [datalust/seq-tickets#2125] [General]
Hi Team,

I am curious to know if we can set custom default values for the time filter and tail feature as a dashboard opens.

In our environment, when a dashboard is opened from the events page, the dashboard time filter is set to "Last 2d". 
Our Seq instance accepts roughly 5k events per minute, and the "Last 2d" filter generally causes a few charts to timeout to prevent i/o exhaustion. Obviously, we could upgrade the hardware running our Seq instance, but typically, while monitoring the system, we only query the "Last 1h" or so in dashboards, and the query performance is quite satisfactory using a shorter time filter. 

It would be great if we could:
1. set a default time filter for each dashboard, and apply it to all users
2. set a default value to enable/disable the tail feature, and apply it to all users
3. allow the user to override time filter and tail after the dashboard opens

This would enable the administrator more easily manage usage of dashboards and allow the dashboard to open with the most appropriate filters. 

Thank you in advance for considering my question.
Scott Mitchell

**Comment by liammclennan** [comment]
Hi Scott,

Thanks for the interesting idea. 5k events per minute is not much so you might be able to get some improvement by designing your dashboards to use signals where possible and not have too many charts. 

You haven't said which version of Seq you are using, but for 2024.1 the dashboard default appears to be 1d. When the range is changed it is stored in localStorage and applied the next time that the dashboard is opened. 

The range and tail settings are stored in the URL, so one way you can set them is to navigate via a URL with pre-set values for `range` and `tail`. The `range` parameter can be any valid Seq duration literal (including things like `range=20m10s`).

**Comment by HarleeDavidson** [comment]
Hi Liam,

We are using version [2023.4.10124]

My reason for asking about setting defaults is that I have users in support who launch the dashboard in the morning and allow it to perpetually refresh all day. They are monitoring ongoing trends/developments in a production system while taking support calls. Having the dashboard already open for reference is important for responding to issues.

However, since they only need to see about 2 hours of history, there's no real reason to go back 2 days.

I've made recommendations to my support team to use a proper date range, and disable the tail feature, but my observations has been that they just open it with the defaults and leave the dashboards open all day.

**Reply by liammclennan** [reply]
The dashboard results are cached, so when the dashboard is reopened, or when tailing is refreshed, only the new values need to be calculated. As long as your Seq server is healthy this should be no problem.

**Comment by jackalvrus** [comment]
Resurrecting this.

It would be very helpful to be able to set the default date range on the Events view.

This is the biggest pain point for me when using custom date ranges. When I use a custom date range, Reset always sets it back to 1 day. Then I have to perform a second operation to select my desired date range.

**Reply by Jaben** [reply]
Agree this is a problem. And you can't use the URL with the range because that breaks default signals.
